Sources/Subs.php

```diff
@@ -6173,4 +6173,21 @@ function build_query_board($userid)
 	return $query_part;
 }
 
+/**
+ * Check if the connection is using https.
+ * 
+ * @return boolean true if connection used https
+ */
+function httpsOn()
+{
+	$secure = false;
+
+	if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') 
+		$secure = true;
+	elseif (!empty($_SERVER['HTTP_X_FORWARDED_PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https' || !empty($_SERVER['HTTP_X_FORWARDED_SSL']) && $_SERVER['HTTP_X_FORWARDED_SSL'] == 'on') 
```

Comment:
Rather than `elseif`, why not just use `||` in the `if`'s condition?

Reply:
first because it's suck to read,
secondly forward stuff is because of load balancer.(to make clear where the https is)

Reply:
Fair enough.